extensions/amp-story-shopping/0.1/test/test-amp-story-shopping-config.js

```diff
@@ -231,5 +223,92 @@ describes.realWin(
         ).to.have.been.calledOnce;
       });
     });
+
+    it('test validate required fields for config', async () => {
+      const invalidConfig = JSON.parse(JSON.stringify(defaultInlineConfig));
+
+      for (const [key, value] of Object.entries(productValidationConfig)) {
+        if (value.includes(validateRequired)) {
+          delete invalidConfig['items'][0][key];
+        }
+      }
+      const spy = env.sandbox.spy(user(), 'warn');
+      await createAmpStoryShoppingConfig(null, invalidConfig);
+      for (const [key, value] of Object.entries(productValidationConfig)) {
+        if (value.includes(validateRequired)) {
+          const errorString = `Error: Field ${key} is required.`;
+          expect(spy).to.have.been.calledWith(
+            'AMP-STORY-SHOPPING-CONFIG',
+            errorString
+          );
+        }
+      }
```

Comment:
You should be able to simplify this by deleting one required field and then expecting that the appropriate error is logged:
```
const requiredKey = 'productId';
delete invalidConfig['items'][0][requiredKey];
...
const errorString = 'Error: Field productId is required.'
```